molgenis-data-vcf/src/main/java/org/molgenis/data/vcf/utils/VcfUtils.java

```diff
@@ -83,337 +78,262 @@ public static String createId(Entity vcfEntity)
-					throw new MolgenisDataException(
-							"Weird situation: we are at sample 1 and want to print FORMAT info but there seems to be none?");
-				}
-
-				// now add SAMPLE data
-				sampleColumn.deleteCharAt(sampleColumn.length() - 1);// delete trailing ':'
-				writer.write(sampleColumn.toString());
+				return newVariant;
+			}
 
-				if (it.hasNext())
+			@Override
+			public Entity next()
+			{
+				Entity variant = null;
+				String peekedId;
+				List<Entity> effectsForVariant = Lists.newArrayList();
+				while (effects.hasNext())
 				{
-					writer.write('\t');
+					peekedId = effects.peek().getEntity(VcfWriterUtils.VARIANT).getIdValue().toString();
+					if (variant == null || variant.getIdValue().toString().equals(peekedId))
+					{
+						Entity effect = effects.next();
+						variant = effect.getEntity(VcfWriterUtils.VARIANT);
+						effectsForVariant.add(effect);
+					}
+					else
+					{
+						Entity newVariant = newVariant(variant, effectsForVariant);
+						return newVariant;
+					}
 				}
+				return newVariant(variant, effectsForVariant);
 			}
-		}
+		};
 	}
 
-	public static boolean checkPreviouslyAnnotatedAndAddMetadata(File inputVcfFile, BufferedWriter outputVCFWriter,
-			List<AttributeMetaData> infoFields) throws MolgenisInvalidFormatException, IOException
+	public static List<Entity> createEntityStructureForVcf(EntityMetaData entityMetaData, String attributeName,
+			Stream<Entity> inputStream)
 	{
-		return checkPreviouslyAnnotatedAndAddMetadata(inputVcfFile, outputVCFWriter, infoFields,
-				Collections.emptyList());
+		return createEntityStructureForVcf(entityMetaData, attributeName, inputStream, Collections.emptyList());
 	}
 
-	/**
-	 * Checks for previous annotations
-	 * 
-	 * @param inputVcfFile
-	 * @param outputVCFWriter
-	 * @param infoFields
-	 * @param attributesToInclude
-	 *            , the AttributeMetaData to write to the VCF file, if empty writes all attributes
-	 * @return
-	 * @throws MolgenisInvalidFormatException
-	 * @throws IOException
-	 */
-	public static boolean checkPreviouslyAnnotatedAndAddMetadata(File inputVcfFile, BufferedWriter outputVCFWriter,
-			List<AttributeMetaData> infoFields, List<String> attributesToInclude)
-					throws MolgenisInvalidFormatException, IOException
+	public static List<Entity> createEntityStructureForVcf(EntityMetaData entityMetaData, String attributeName,
+			Stream<Entity> inputStream, List<AttributeMetaData> annotatorAttributes)
 	{
-		String checkAnnotatedBeforeValue = attributesToInclude.isEmpty()
-				? (infoFields.isEmpty() ? null : infoFields.get(0).getName()) : attributesToInclude.get(0);
-		boolean annotatedBefore = false;
+		AttributeMetaData attributeToParse = entityMetaData.getAttribute(attributeName);
+		String description = attributeToParse.getDescription();
+		String[] step1 = description.split(":");
+		String entityName = org.apache.commons.lang.StringUtils.deleteWhitespace(step1[0]);
+		String value = step1[1].replaceAll("^\\s'|'$", "");
```

Comment:
There is no step[0] or step [1] id description is null or hes no ":" can you check this?